File: python/프로세스.py
def solution(priorities, location):
    answer = [0] * size
    priorities = [(i, p) for i, p in enumerate(priorities)]
    first = max(priorities, key=lambda x: x[1])[1]

    cnt = 1
    while len(priorities) > 1:
        idx, priority = priorities.pop(0)
        if priority >= first:
            answer[idx] = cnt
            cnt += 1
            first = max(priorities, key=lambda x: x[1])[1]
        else:
            priorities.append((idx, priority))
    
    answer[priorities[0][0]] = cnt
    
    return answer[location]



# --------------------------------------------
# any() 사용

def solution(priorities, location):
    answer = 0
    queue = [(i, p) for i, p in enumerate(priorities)]

    while True:
        idx, priority = queue.pop(0)
        if any(priority < q[1] for q in queue):
            queue.append((idx, priority))
        else:
            answer += 1
            if idx == location:
                break
    
    return answer
    

File: java/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        // 내림차순 우선순위 큐
        PriorityQueue<Integer> q = new PriorityQueue<>(Collections.reverseOrder());
        
        for(int priority: priorities){
            q.offer(priority);
        }
        
        while(!q.isEmpty()){
            for(int i=0; i<priorities.length; i++){
                if (q.peek() == priorities[i]) {
                    q.poll();
                    answer++;

                    // 현재 작업이 location과 같으면 answer 반환
                    if (location == i) {
                        return answer;
                    }
                }
            }
        }
        
        return answer;
    }
}
